### PR TITLE
modified question mark query

### PR DESCRIPTION
**Pull request by jshen1s1**
for all filters: /api/department?dep=any&type=any&ge=any
for single class: /api/course?course=XX

### CODE REVIEW

**Review by colinaguan — COMMENTED** [review]
always use === and !== in javascript

File: backend/routes/department.js [hunk 1]
```diff
@@ -49,12 +49,43 @@ router.get("/", function (req, res) {
           }
         }
       }
+    // case type=XX
+    } else if (queryParameter.type && queryParameter.type !== 'any') {
+      for (var i in classData){
+        for (var j in classData[i]){
+            if (classData[i][j].type == queryParameter.type.toUpperCase()){
```
Comment (colinaguan) [inline backend/routes/department.js]: === in javascript

File: backend/routes/department.js [hunk 2]
```diff
@@ -49,12 +49,43 @@ router.get("/", function (req, res) {
...
+                // case type=xx&ge=xx
+                if (queryParameter.ge && queryParameter.ge !== 'any') {
+                    for (var k in classData[i][j].ge){
+                        if (classData[i][j].ge[k] == queryParameter.ge.toUpperCase()){
```
Comment (colinaguan) [inline backend/routes/department.js]: ===

File: backend/routes/department.js [hunk 3]
```diff
@@ -49,12 +49,43 @@ router.get("/", function (req, res) {
...
+                }
+            }
+        }
+      }
+    //case ge=XX
+    } else if (queryParameter.ge && queryParameter.ge !== 'any'){
+      for (var i in classData){
+          for (var j in classData[i]){
+              for (k in classData[i][j].ge){
+                  if (classData[i][j].ge[k] == queryParameter.ge.toUpperCase()){
```
Comment (colinaguan) [inline backend/routes/department.js]: ===

File: backend/routes/course.js [hunk 4]
```diff
@@ -9,16 +9,14 @@ router.get('/', function(req, res) {
     
     // if there are parameters entered
     if (Object.keys(req.query).length != 0) {
```
Comment (colinaguan) [inline backend/routes/course.js]: !==

**Review by colinaguan — CHANGES_REQUESTED** [review]
there's a lot of != and == in here (only commented on a few) but just make sure all of them are replaced

one of the reasons why javascript is annoying is because == vs. === can be the difference between a space shuttle crashing into the moon or landing successfully... yay javascript

**Review by colinaguan — APPROVED** [review]

**Review by Brenden-Greenlee — APPROVED** [review]

**Pull request merged**